requirements-ext.txt

```diff
@@ -1,7 +1,6 @@
 cached_property
 cachetools
 ipython
-matplotlib
```

Comment:
This needs to be added as an option to `firedrake-install` somehow, `--no-matplotlib`?

Reply:
Sure

Reply:
Did you change your mind on this point?

Reply:
I think it's possibly quite reasonable to not install it by default and prompt the users to do so when they hit the import error. Not sure though. It would keep the install script a little simpler.

Reply:
I'm happy with this line getting deleted so long as there is a meaningful message when users hit an MPL `ImportError` prompting them to `pip install`.

Reply:
I am going to argue that the import error that a user gets when trying to use `matplotlib` is sufficient here.

Reply:
I'm not sure about that. I think users may see it as a bug if they try `import firedrake.pyplot` and it fails. If we have an explicit message about requiring a pip install then they will see that it's deliberate.